src/contexts/auth-context.js

```diff
@@ -0,0 +1,39 @@
+import { React, useContext, createContext, useState, useEffect } from "react";
+import axios from "axios";
+import { Navigate, useNavigate } from "react-router-dom";
+
+const authContext = createContext();
+
+const AuthProvider = ({ children }) => {
+  const [user, setUser] = useState();
+  const [encodedToken, setEncodedToken] = useState();
+  const [isLoggedIn, setIsLoggedIn] = useState(false);
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    const fetch = async () => {
+      const userData = localStorage.getItem("user");
+      const value = await axios.post("/api/auth/signup", userData);
+      setEncodedToken(value.data.encodedToken);
+    };
+    fetch();
+  }, [encodedToken]);
+
+  const logOut = () => {
+    setIsLoggedIn(false);
+    localStorage.removeItem("token");
+    navigate("/");
+  };
+
+  return (
+    <authContext.Provider
```

Comment:
in `authContext`,  'A' should be in capital  letters

Reply:
done